qemu_kvm.go

```diff
@@ -0,0 +1,419 @@
+//
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package virtcontainers
+
+import (
+	"encoding/hex"
+	"fmt"
+	"os"
+
+	govmmQemu "github.com/intel/govmm/qemu"
+)
+
+type qemuArch interface {
+	// runningOnVMM returns true if the system is running inside a VM
+	runningOnVMM(cpuInfoPath string) (bool, error)
```

Comment:
is not qemu specific, but it's arch specific, in ARM is not possible to detect it

Reply:
Right now it's not because nested virt in ARM is not very mature, but I expect at some point we will be able to detect it in an hypervisor agnostic way. So it should not be a qemu specific method if we want e.g. to support other hypervisors and not duplicate that code.

Reply:
ya, make sense